form-data/form-data.d.ts

```diff
@@ -1,16 +1,31 @@
 // Type definitions for form-data
 // Project: https://github.com/felixge/node-form-data
-// Definitions by: Carlos Ballesteros Velasco <https://github.com/soywiz>
+// Definitions by: Carlos Ballesteros Velasco <https://github.com/soywiz>, Leon Yu <https://github.com/leonyu>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
 // Imported from: https://github.com/soywiz/typescript-node-definitions/form-data.d.ts
 
+declare namespace NodeFormData {
```

Comment:
First off, thanks @leonyu for your contribution.  I appreciate the addition of the code examples from the project into the tests.  I do however have a concern about this PR. Why is the `NodeFormData` namespace declared in the global scope?  I assume it's the way you avoided the error when using ES6 import syntax, but it does cause some other problems.
In the request-tests below, it is more difficult to see where it is coming from because you are no longer importing the type, it's just there.  Also, the person consuming the types can't rename the type on import, since it's declared in the global scope, which doesn't really reflect reality as `NodeFormData` won't actually exist on the global scope.
IMHO, there is a better alternative way of making the imports work without polluting the global scope.  You could do something like this:

```ts
declare module "form-data" {
	class FormData {
		append(key: string, value: any, options?: any): void;
		getHeaders(): FormData.Dictionary<string>;
		// TODO expand pipe
		pipe(to: any): any;
		submit(params: string | Object, callback: (error: any, response: any) => void): any;
		getBoundary(): string;
	}

    namespace FormData {
        interface Dictionary<T> {
            [key: string]: T;
        }
    }

    export = FormData;
}

```

This allows for these imports to work: 

```ts
import FormData = require('form-data');
import * as FormData2 from 'form-data';
```

But you no longer have to worry about global scope pollution.